crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,131 @@
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::{
+    runtime_api::api, submit_extrinsic, Client, CodeHash, ContractMessageTranscoder,
+    DefaultConfig, ExtrinsicOpts, PairSigner, TokenMetadata,
+};
+use crate::{
+    cmd::extrinsics::{events::DisplayEvents, parse_code_hash, ErrorVariant},
+    name_value_println,
+};
+use anyhow::{Ok, Result};
+use std::fmt::Debug;
+use subxt::{Config, OnlineClient};
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "remove", about = "Remove a contract's code")]
+pub struct RemoveCommand {
+    /// The hash of the smart contract code already uploaded to the chain.
+    #[clap(long, value_parser = parse_code_hash)]
+    code_hash: Option<<DefaultConfig as Config>::Hash>,
+    #[clap(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl RemoveCommand {
+    pub fn is_json(&self) -> bool {
+        self.output_json
+    }
+
+    pub fn run(&self) -> Result<(), ErrorVariant> {
+        let artifacts = self.extrinsic_opts.contract_artifacts()?;
+        let transcoder = artifacts.contract_transcoder()?;
+        let signer = super::pair_signer(self.extrinsic_opts.signer()?);
+
+        let artifacts_path = artifacts.artifact_path().to_path_buf();
+
+        let final_code_hash = match (self.code_hash.as_ref(), artifacts.code.as_ref()) {
+            (Some(code_hash), Some(_)) => {
+                code_hash;
+                Ok(())
+            }
+            (Some(code_hash), None) => {
+                code_hash;
+                Ok(())
+            }
+            (None, Some(_)) => {
+                artifacts.code_hash()?;
+                Ok(())
+            }
+            (None, None) => {
+                Err(
+                anyhow::anyhow!(
+                "No code_hash was provided or contract code was not found from artifact file {}. Please provide a code hash with --code-hash argument or specify the path for artifacts files with --manifest-path", 
+                artifacts_path.display()
+            ).into())
+            }
+        };
+
+        async_std::task::block_on(async {
+            let url = self.extrinsic_opts.url_to_string();
+            let client = OnlineClient::from_url(url.clone()).await?;
+
+            if let Some(code_removed) = self
+                .remove_code(
+                    &client,
+                    sp_core::H256(final_code_hash),
+                    &signer,
+                    &transcoder,
+                )
+                .await?
+            {
+                let remove_result = code_removed.code_hash;
+
+                if self.output_json {
+                    println!("{}", remove_result);
+                } else {
+                    name_value_println!("Code hash", format!("{:?}", remove_result));
+                }
+                Ok(())
```

Comment:
```suggestion
                Result::<(), ErrorVariant>::Ok(())
```
This will fix one of your errors. It's not particularly nice, but it gives you a clue that you need to give a type hint for it to infer the error type of the `Result` here. So see if you can find a more elegant way to do this.

Reply:
Thanks